api/src/controllers/user.js

```diff
@@ -85,124 +86,142 @@ exports.list = (req, res) => {
 	}
 };
 
-exports.delete = (req, res) => {
+exports.delete = async (req, res) => {
 	// authorize access to the authenticated user model only
 	if (req.params.userId !== req.user.sub) {
-		return res.status(401).send();
+		return res.status(403).send();
 	}
 
 	// check to see if the user exists
-	User.findById(req.params.userId)
-		.then(user => {
-			if (!user) {
-				// corner case
-				return res.sendStatus(404);
-			} else {
-				// remove the user document; 204 No Content confirms success
-				return User.findOneAndRemove({ _id: req.params.userId })
-					.then(user => res.status(204).send());
-			}
-		})
-		.catch(err => {
-			logger.error(err);
-			res.status(500).send(err);
-		});
+	let user;
+	try {
+		user = await User.findById(req.params.userId);
+	} catch(error) {
+		logger.error(err);
+		res.status(500).send(err);
+		return;
+	}
+
+	if (!user) {
+		res.sendStatus(404);
+		return;
+	}
+
+	// remove the user document; 204 No Content confirms success
+	try {
+		await User.findOneAndRemove({ _id: req.params.userId });
```

Comment:
can drop the try/catch. I suggest adding a method to `User` to cascade deletes to related documents in other collections (eg. Pins, Follows, Listen)